src/test/java/org/terracotta/offheapstore/AbstractOffHeapMapIT.java

```diff
@@ -74,6 +75,34 @@ public AbstractOffHeapMapIT(Generator generator) {
   protected abstract Map<Integer, byte[]> createOffHeapBufferMap(PageSource source);
 
   @Test
+  public void testRemoveByHash() throws Exception {
+    final int ENTRIES_COUNT = 1013;
+    Map<SpecialInteger, SpecialInteger> map = createMap(generator);
+    if (!(map instanceof HashingMap)) {
+      throw new AssertionError("map is instance of " + map.getClass());
+    }
+    HashingMap<SpecialInteger, SpecialInteger> offHeapHashMap = (HashingMap<SpecialInteger, SpecialInteger>) map;
+
+    for (int i = 0; i < ENTRIES_COUNT; i++) {
+      SpecialInteger generated = generator.generate(i);
+      offHeapHashMap.put(generated, generated);
+    }
+
+    Map<SpecialInteger, SpecialInteger> removed = offHeapHashMap.removeByHash(BadInteger.HASHCODE);
+    assertThat(removed.size() > 0, is(true));
+    assertThat(offHeapHashMap.size(), is(ENTRIES_COUNT - removed.size()));
+
+    for (int i = 0; i < ENTRIES_COUNT; i++) {
+      SpecialInteger generated = generator.generate(i);
+      if (removed.containsKey(generated)) {
+        assertThat(offHeapHashMap.get(generated), is(nullValue()));
+      } else {
+        assertThat(offHeapHashMap.get(generated), equalTo(generated));
```

Comment:
no need to import equalTo... we're already using is(...) for this purpose:
`assertThat(offHeapHashMap.get(generated), is(generated));`